docs/cross-chain-payments-and-execution.md

```diff
@@ -0,0 +1,77 @@
+## Cross chain payments and execution
+
+This document suggests an approach of using `UserOp`s to perform cross-chain payments and execution. At a very high level this approach can be thought of as a cross-chain state channel, where we have some state, that when fully signed can be submitted to an adjudicator contract to claim the funds based on the outcome of that state.
+
+However instead of implementing the state channel framework ourselves, we're making use of some of ERC 4337 infrastructure. Instead of signing a state, participants sign a **UserOp** that contains the state information. Instead of having a specific "adjudicator" contract, we have the **entrypoint** and the **BridgeWallet SCW** to handle adjudicating funds. Instead of making a on-chain call to challenge on the adjudicator, we can submit the `UserOp` to an entrypoint to trigger a challenge.
+
+Since we can specify the UserOp calldata, we can include whatever additional state information we want in that calldata. That lets us embed payment or execution information for different chains in one UserOp. When submitted to chain the UserOp will behave depending on the chain Id and the embedded payment or execution information.
+
+# Payments
+
+We embed payment information in the UserOp using a `PaymentChainInfo`
+
+```sol
+/// Contains information to execute a payment
+struct PaymentChainInfo {
+  uint chainId;
+  address entrypoint;
+  State paymentState;
+}
+
+```
+
+Whenever a fully signed UserOp containing a `PaymentChainInfo` is submitted to chain, it forces a challenge on that chain. This means that once you have a fully signed UserOp you know you can always use it to get your funds via challenge. In the happy path, we would expect participants to just exchange signed states afterwards, so they can discard the UserOp.
```

Comment:
Users can always submit the signed state directly to initiate a challenge, so it should be safe to discard the UserOp after everyone signs the state.

Reply:
Ok great. It might be nice to include some details indicating users can challenge using just the signed state (and maybe how that is done) and that is why it is safe to discard the signed `UserOp`